tests/ui/panic-handler/panic-handler-bad-signature-2.stderr

```diff
@@ -1,8 +1,18 @@
-error: argument should be `&PanicInfo`
-  --> $DIR/panic-handler-bad-signature-2.rs:10:11
+error[E0308]: `#[panic_handler]` function has wrong type
+  --> $DIR/panic-handler-bad-signature-2.rs:9:1
    |
-LL |     info: &'static PanicInfo,
-   |           ^^^^^^^^^^^^^^^^^^
+LL | fn panic(info: &'static PanicInfo) -> !
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
+   |
+   = note: expected signature `fn(&PanicInfo<'_>) -> _`
+              found signature `fn(&'static PanicInfo<'_>) -> _`
+note: the anonymous lifetime as defined here...
```

Comment:
This is pretty confusing. We probably want to actually replace the call to `ocx.resolve_regions_and_report_errors` to a manual call to `ocx.infcx.resolve_regions` and report any region errors manually from there with a more descriptive message.

Reply:
What message would we want here?

In the case of a trait/impl:
```rust
use std::panic::PanicInfo;

trait Tr {
    fn panic(info: &PanicInfo<'_>);
}

impl Tr for () {
    fn panic(_info: &'static PanicInfo) {
        unimplemented!();
    }
}
```
It is reported similarly:
```
error[E0308]: method not compatible with trait
 --> lib.rs:8:5
  |
8 |     fn panic(_info: &'static PanicInfo) {
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime mismatch
  |
  = note: expected signature `fn(&PanicInfo<'_>)`
             found signature `fn(&'static PanicInfo<'_>)`
note: the anonymous lifetime as defined here...
 --> lib.rs:8:30
  |
8 |     fn panic(_info: &'static PanicInfo) {
  |                              ^^^^^^^^^
  = note: ...does not necessarily outlive the static lifetime

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
```

The difference is that in the test, the note spans the whole signature instead of just `PanicInfo`:
```
note: the anonymous lifetime as defined here...
  --> $DIR/panic-handler-bad-signature-2.rs:9:1
   |
LL | fn panic(info: &'static PanicInfo) -> !
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: ...does not necessarily outlive the static lifetime
```
I'm not sure why.

Reply:
I opened https://github.com/rust-lang/rust/issues/115903 regarding confusing lifetime diagnosis for trait impl functions. I think fixing that its outside the scope of this PR.